pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -85,6 +85,10 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
       if (csvHeader == null) {
         format = format.withHeader();
       } else {
+        //we assume that the record delimiter also delimits the header
+        if (delimiterNotPresentInHeader(delimiter, csvHeader)) {
```

Comment:
an edge case that we should prolly handle - what if the record contains a single value? This is quite possible especially if users have a long string that they want to split up into multiple fields using ingestion transforms or use json index on it.